lib/gradebook.rb

```diff
@@ -1 +1,52 @@
+class Gradebook
+  attr_reader :instructor, :courses, :all_students
+  def initialize (instructor)
+    @instructor = instructor
+    @courses = []
+    @all_students_a
+  end
 
+  def add_course(course)
+    @courses << course
+  end
+
+  def all_students
+    all_students = []
+    @courses.each {|course| all_students << course.students}
+    all_students.flatten
```

Comment:
`.each` works, but you could also use [flat_map](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-flat_map)
```suggestion
    @courses.flat_map {|course| course.students}
```